### PR TITLE
[inner-1555] manual kill cluster renew thread

### PR DESCRIPTION
**Pull request by wenyh1**
Reason:  
  Improve inner 1555
Type:  
  BUG/Improve  
Influences：  
   fix xx

### CODE REVIEW

File: version.txt [hunk 1]
```diff
@@ -1,5 +1,5 @@
-GitVersion f5928d276537752e7a4760aac6193f5e48531c36
-BuildTime  20220524130848
+GitVersion 65b108afa613b6d8095aded72cf93ff537ff7319
+BuildTime  20220608144216
```
Comment (dcy10000) [inline version.txt:2]: why update this?

**Review by dcy10000 — CHANGES_REQUESTED** [review]

File: src/main/java/com/actiontech/dble/cluster/general/AbstractConsulSender.java [hunk 2]
```diff
@@ -157,5 +164,26 @@ protected void firstReturnToCluster() throws Exception {
         }
     }
 
+    public List<String> fetchRenewThread() {
+        String onlineStr = ClusterPathUtil.getOnlinePath(SystemConfig.getInstance().getInstanceName());
+        List<String> renewThread = Lists.newArrayList();
+        lockMap.values().stream().
+                filter(c -> !c.getName().endsWith(onlineStr) && c.isAlive()).
+                forEach(c -> renewThread.add(c.getName()));
```
Comment (dcy10000) [inline src/main/java/com/actiontech/dble/cluster/general/AbstractConsulSender.java:172]: use .collect(toList()) instead is better

File: src/main/java/com/actiontech/dble/cluster/general/AbstractConsulSender.java [hunk 3]
```diff
@@ -157,5 +164,26 @@ protected void firstReturnToCluster() throws Exception {
...
+    }
 
+    public boolean killRenewThread(String path) {
+        Thread renewThread = lockMap.get(path);
+        if (renewThread != null) {
+            if (renewThread.isAlive() && !renewThread.isInterrupted()) {
+                renewThread.interrupt();
+                LOGGER.info("manual kill cluster renew thread: [{}]", renewThread.getName());
+            } else {
+                LOGGER.info("try manual kill cluster renew thread: [{}], but it already killed", renewThread.getName());
```
Comment (dcy10000) [inline src/main/java/com/actiontech/dble/cluster/general/AbstractConsulSender.java:183]: maybe interrupted but not killed?

File: src/main/java/com/actiontech/dble/services/manager/response/KillClusterRenewThread.java [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+import com.actiontech.dble.net.mysql.OkPacket;
+import com.actiontech.dble.services.manager.ManagerService;
+import com.actiontech.dble.util.StringUtil;
+
+public final class KillClusterRenewThread {
+    private KillClusterRenewThread() {
+    }
+
+    public static void response(ManagerService service, String name) {
+        String str = getResult(name);
```
Comment (dcy10000) [inline src/main/java/com/actiontech/dble/services/manager/response/KillClusterRenewThread.java:15]: shouldn't always return ok.

**Review by dcy10000 — CHANGES_REQUESTED** [review]

File: src/main/java/com/actiontech/dble/cluster/general/AbstractConsulSender.java (hunk 2 above)
Comment (wenyh1) [reply src/main/java/com/actiontech/dble/cluster/general/AbstractConsulSender.java]: ok

File: src/main/java/com/actiontech/dble/services/manager/response/KillClusterRenewThread.java (hunk 4 above)
Comment (wenyh1) [reply src/main/java/com/actiontech/dble/services/manager/response/KillClusterRenewThread.java]: ok

File: src/main/java/com/actiontech/dble/cluster/general/AbstractConsulSender.java (hunk 3 above)
Comment (wenyh1) [reply src/main/java/com/actiontech/dble/cluster/general/AbstractConsulSender.java]: ok

**Review by dcy10000 — APPROVED** [review]
ok

**Pull request merged**